code/obj/critter/misc.dm

```diff
@@ -1040,27 +1040,33 @@
 		src.attacking = 1
 		src.visible_message("<span class='combat'><B>The [src.name]</B> starts to envelop [M]!</span>")
 
-		var/lastloc = M.loc
-		SPAWN(6 SECONDS)
-			if (get_dist(src, M) <= 1 && ((M.loc == lastloc)))
-				if(isliving(M))
-					logTheThing("combat", M, null, "was enveloped by [src] (obj) at [log_loc(src)].") // Some logging for instakill critters would be nice (Convair880).
-					M.ghostize()
-
-					if (iscarbon(M))
-						for(var/obj/item/W in M)
-							if (isitem(W))
-								M.u_equip(W)
-								if (W)
-									W.set_loc(M.loc)
-									W.dropped(M)
-									W.layer = initial(W.layer)
-
-				src.visible_message("<span class='combat'><B>The [src.name]</B> completely envelops [M]!</span>")
-				playsound(src.loc, "sound/impact_sounds/Slimy_Hit_4.ogg", 50, 1)
-				qdel(M)
+		SETUP_GENERIC_ACTIONBAR(src, src, 6 SECONDS, .proc/finish_envelope, list(M), 'icons/mob/critter_ui.dmi', "devour_over", \
+					"", INTERRUPT_ACTION | INTERRUPT_MOVE | INTERRUPT_STUNNED | INTERRUPT_ACT)
```

Comment:
Can't you abuse `INTERRUPT_MOVE` with pull?
 
Consider checking distance similar to mob ability or dropping flag to maintain behavior.

Reply:
I'm not really sure what you mean by this. As far as I can tell, pulling the ghost doesn't interrupt the killing unless you (as the target) actually move out of range, causing the ghost to move to catch you. You don't seem to be able to otherwise interrupt it.